pkg/storage/mysql/mysqlstore.go

```diff
@@ -0,0 +1,268 @@
+	p.Lock()
+	defer p.Unlock()
+
+	for _, store := range p.dbs {
+		err := store.db.Close()
+		if err != nil {
+			return fmt.Errorf(failToCloseProviderErrMsg+": %w", err)
+		}
+	}
+
+	if err := p.db.Close(); err != nil {
+		return err
+	}
+
+	p.dbs = make(map[string]*sqlDBStore)
+
+	return nil
+}
+
+// CloseStore closes a previously opened store
+func (p *Provider) CloseStore(name string) error {
+	p.Lock()
+	defer p.Unlock()
+
+	store, exists := p.dbs[name]
+	if !exists {
+		return nil
+	}
+
+	delete(p.dbs, name)
+
+	return store.db.Close()
+}
+
+// Put stores the key and the record
+func (s *sqlDBStore) Put(k string, v []byte) error {
+	if k == "" {
+		return errors.New("key is mandatory")
+	}
+
+	//nolint: gosec
+	// create upsert query to insert the record, checking whether the key is already mapped to a value in the store.
+	createStmt := "INSERT INTO " + s.tableName + " VALUES (?, ?) ON DUPLICATE KEY UPDATE value=?"
+	// executing the prepared insert statement
+	_, err := s.db.Exec(createStmt, k, v, v)
+	if err != nil {
+		return fmt.Errorf("failed to insert key and value record into %s %w ", s.tableName, err)
+	}
+
+	return nil
+}
+
+// Get fetches the record based on key
+func (s *sqlDBStore) Get(k string) ([]byte, error) {
+	if k == "" {
+		return nil, errors.New("key is mandatory")
+	}
+
+	var value []byte
+	//nolint: gosec
+	// select query to fetch the record by key
+	err := s.db.QueryRow("SELECT `value` FROM "+s.tableName+" "+
+		" WHERE `key` = ?", k).Scan(&value)
+	if err != nil {
+		if strings.Contains(err.Error(), sqlDBNotFound) {
+			return nil, storage.ErrValueNotFound
+		}
+
+		return nil, fmt.Errorf("failed to get row %w", err)
+	}
+
+	return value, nil
+}
+
+type sqlDBResultsIterator struct {
+	resultRows *sql.Rows
+	err        error
+}
+
+func (s *sqlDBStore) CreateIndex(createIndexRequest storage.CreateIndexRequest) error {
+	// TODO: Issue-36 Implement Create Index
+	return nil
+}
+
+func (s *sqlDBStore) Query(findQuery string) (storage.ResultsIterator, error) {
+	params := &queryParams{}
+
+	err := json.Unmarshal([]byte(findQuery), params)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal the query string %w", err)
+	}
+
+	if strings.Contains(params.EndKey, storage.EndKeySuffix) {
+		params.EndKey = strings.ReplaceAll(params.EndKey, storage.EndKeySuffix, "*")
+	}
+	//nolint:gosec
+	// sub query to fetch the all the keys that have start and end key reference, simulating range behavior.
+	queryStmt := "SELECT * FROM " + s.tableName + " WHERE `key` >= ? AND `key` < ? order by `key`"
+
+	resultRows, err := s.db.Query(queryStmt, params.StartKey, params.EndKey)
+	if err != nil && resultRows == nil {
+		return &sqlDBResultsIterator{
+			err: fmt.Errorf("failed to query rows %w", err)}, err
+	} else if err := resultRows.Err(); err != nil {
+		return &sqlDBResultsIterator{
+			err: fmt.Errorf("failed to get resulted rows %w", err)}, err
+	}
+
+	return &sqlDBResultsIterator{resultRows: resultRows}, nil
+}
```

Comment:
Query is supposed to be very generic - basically it should be possible to pass in the entire query here. It looks like your logic here only allows certain types of queries, which won't really work for `edv` - take a look at the `Query` method in `couchdbstore.go`

Reply:
@llorllale I am not sure how this will reflect in mysql, Can you suggest what we are looking for in mysql query implementation ? Couchdb provides https://docs.couchdb.org/en/stable/api/database/find.html find endpoint ,

Reply:
updated logic to accept any query rather than doing specific